engine/src/main/java/org/terasology/rendering/nui/layers/ingame/metrics/DebugRenderingSystem.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.rendering.nui.layers.ingame.metrics;
+
+
+import com.google.common.collect.Iterators;
+import com.google.common.collect.Sets;
+import org.terasology.entitySystem.systems.BaseComponentSystem;
+import org.terasology.entitySystem.systems.RegisterSystem;
+import org.terasology.registry.Share;
+
+import java.util.Iterator;
+import java.util.LinkedHashSet;
+
+@RegisterSystem
+@Share(DebugRenderingSystem.class)
+public class DebugRenderingSystem extends BaseComponentSystem {
+
+    private LinkedHashSet<MetricsMode> metricsModes;
+    private Iterator<MetricsMode> modeIterator;
+    private MetricsMode currentMode;
+
+    @Override
+    public void initialise() {
+        metricsModes = Sets.newLinkedHashSet();
+        modeIterator = Iterators.cycle(metricsModes);
+        currentMode = register(new NullMetricsMode());
+        register(new RunningMeansMode());
+        register(new SpikesMode());
+        register(new AllocationsMode());
+        register(new RunningThreadsMode());
+        register(new WorldRendererMode());
+        register(new NetworkStatsMode());
+        register(new RenderingExecTimeMeansMode("Rendering - Execution Time: Running Means - Sorted Alphabetically"));
+    }
+
+    public MetricsMode register(MetricsMode mode) {
```

Comment:
I had a hunch that something wasn't quite right with this class but I couldn't quite point the finger to it. I now realize that issue #1773 recommends to add register() and unregister() methods to the DebugOverlay class. That's because DebugOverlay is not intended to be rendering-specific. Systems or even modules might wish to register their own MetricsMode with it, completely unrelated to rendering.

This is also in line with the NetworkStatsMode() above not being quite rendering-related. It is true that at this stage the DebugOverlay is mostly used for rendering but it's not intended to be exclusive for it.

So, I recommend to remove this class and move its functionality into DebugOverlay. I.e. I like the switch to a LinkedHashSet and the use of register() to register the default Metrics. 

Eventually we should move the registration of the default rendering metrics into the WorldRenderer initialization.